<launch>
	<!-- merge two LaserScan topics -->
	<!-- <node pkg="topic_tools" type="relay" name="relay_tilt_scan_to_base_scan" args="tilt_scan base_scan" output="screen" /> -->


	<!-- LaserScan assembler -->
	<node pkg="laserscan_to_pointcloud" type="laserscan_to_pointcloud_assembler" name="laserscan_to_pointcloud_assembler" output="screen">
		<param name="laser_scan_topic" type="str" value="base_scan" />
		<param name="pointcloud_publish_topic" type="str" value="planar_pointcloud" />
		<param name="number_of_scans_to_assemble_per_cloud" type="int" value="1" />
		<param name="timeout_for_cloud_assembly" type="double" value="1.0" />
		<param name="target_frame" type="str" value="map" />
		<param name="min_range_cutoff_percentage_offset" type="double" value="5.00" />
		<param name="max_range_cutoff_percentage_offset" type="double" value="0.95" />
		<param name="include_laser_intensity" type="bool" value="false" />
		<param name="interpolate_scans" type="bool" value="false" />
	</node>


	<!-- map to odom publisher -->
	<include file="$(find dynamic_robot_localization)/launch/pose_to_tf_publisher.launch" />

	<!-- planar localization -->
	<node pkg="dynamic_robot_localization" type="planar_localization_node" name="planar_localization_node" output="screen">
		<!-- localization params -->
		<param name="planar_pointcloud_topic" type="str" value="planar_pointcloud" />
		<param name="costmap_topic" type="str" value="map" />
		<param name="reference_map_pointcloud_publish_topic" type="str" value="reference_map_pointcloud" />
		<param name="aligned_pointcloud_publish_topic" type="str" value="aligned_pointcloud" />
		<param name="pose_publish_topic" type="str" value="initialpose" />
		<param name="publish_tf_map_odom" type="bool" value="false" />
		<param name="base_link_frame_id" type="str" value="base_footprint" />
		<param name="min_seconds_between_scan_registration" type="double" value="0.05" />
		<param name="min_seconds_between_map_update" type="double" value="5.0" />
		<param name="max_seconds_scan_age" type="double" value="0.2" />

		<!-- ICP params -->
		<param name="max_alignment_fitness" type="double" value="1e-2" />
		<param name="max_transformation_angle" type="double" value="1.59" />
		<param name="max_transformation_distance" type="double" value="2.5" />
		<param name="max_correspondence_distance" type="double" value="2.5" />
		<param name="transformation_epsilon" type="double" value="1e-8" />
		<param name="euclidean_fitness_epsilon" type="double" value="1e-6" />
		<param name="max_number_of_registration_iterations" type="int" value="250" />
		<param name="max_number_of_ransac_iterations" type="int" value="100" />
		<param name="ransac_outlier_rejection_threshold" type="double" value="0.05" />
	</node>


	<!-- localization tests -->
	<include file="$(find dynamic_robot_localization)/tests/localization_plots.launch" />
</launch>